core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -1474,6 +1474,27 @@ private void basePushFilterPastAggWithGroupingSets(boolean unchanged)
     sql(sql).with(program).check();
   }
 
+  @Test public void testDistinctWithFilterWithoutGroupBy() {
+    final String sql = "SELECT SUM(comm), COUNT(DISTINCT comm),\n"
+        + "COUNT(DISTINCT sal) FILTER (WHERE sal > 1000)\n"
+        + "FROM emp";
+    HepProgram program = new HepProgramBuilder()
+        .addRuleInstance(AggregateExpandDistinctAggregatesRule.INSTANCE)
+        .build();
+    sql(sql).with(program).check();
+  }
+
+  @Test public void testDistinctWithFilterAndGroupBy() {
+    final String sql = "SELECT deptno, SUM(comm), COUNT(DISTINCT comm),\n"
+        + "COUNT(DISTINCT sal) FILTER (WHERE sal > 1000)\n"
```

Comment:
Is this a minimal reproducible example?

Frankly speaking, it is really hard to follow the code, and it is sad it is the most trivial example in the PR.

I would prefer to have very simple example that makes it clear what the rule produces.

Reply:
Hi, vlsi, the mininal reproducible case is there exists two aggregation call, one is distinct aggregate call with filter, the other can be distinct agg call or not. I've update the JIRA description with a comprehensible case. And the test cases here can be simplified a little actually.